tests/tasks/zendesk/test_zendesk_tasks.py

```diff
@@ -0,0 +1,251 @@
+                api_token="abc", subdomain="foo", email_address="foo@bar.com"
+            )
+
+        assert "Both `start_time` and `cursor` are missing." in str(exc)
+
+    @responses.activate
+    def test_run_with_cursor(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        zendesk_task = ZendeskTicketsIncrementalExportTask()
+        responses.add(
+            responses.GET,
+            url="https://test.zendesk.com/api/v2/incremental/tickets/cursor.json?cursor=xyz",
+            json={
+                "end_of_stream": True,
+                "after_url": "foo",
+                "after_cursor": "foo",
+                "tickets": [],
+            },
+            headers={"retry-after": "1"},
+            status=200,
+        )
+
+        zendesk_task.run(
+            subdomain="test", email_address="foo@bar.com", api_token="abc", cursor="xyz"
+        )
+
+        assert "Got cursor" in caplog.text
+
+    @responses.activate
+    def test_run_with_start_time(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        zendesk_task = ZendeskTicketsIncrementalExportTask()
+        responses.add(
+            responses.GET,
+            url="https://test.zendesk.com/api/v2/incremental/tickets/cursor.json?start_time=123",
+            json={
+                "end_of_stream": True,
+                "after_url": "foo",
+                "after_cursor": "foo",
+                "tickets": [],
+            },
+            headers={"retry-after": "1"},
+            status=200,
+        )
+
+        zendesk_task.run(
+            subdomain="test",
+            email_address="foo@bar.com",
+            api_token="abc",
+            start_time=123,
+        )
+
+        assert "Got start_time" in caplog.text
+
+    @responses.activate
+    def test_run_retry_after(self, caplog):
+        caplog.set_level(logging.DEBUG)
+        zendesk_task = ZendeskTicketsIncrementalExportTask()
+        responses.add(
+            responses.GET,
+            url="https://test.zendesk.com/api/v2/incremental/tickets/cursor.json?start_time=123",
+            json={
+                "end_of_stream": False,
+                "after_url": "foo",
+                "after_cursor": "foo",
+                "tickets": [],
+            },
+            headers={"retry-after": "1"},
+            status=429,
+        )
+
+        responses.add(
+            responses.GET,
+            url="https://test.zendesk.com/api/v2/incremental/tickets/cursor.json?start_time=123",
+            json={
+                "end_of_stream": True,
+                "after_url": "foo",
+                "after_cursor": "foo",
+                "tickets": [],
+            },
+            headers={"retry-after": "1"},
+            status=200,
+        )
+
+        zendesk_task.run(
+            subdomain="test",
+            email_address="foo@bar.com",
+            api_token="abc",
+            start_time=123,
+        )
+
+        assert "API rate limit reached!" in caplog.text
```

Comment:
Tests that check what is logged by the task are coupled to the specific log message. In order to make the test more robust, you could use the context manager implementation provided by responses (https://github.com/getsentry/responses#responses-as-a-context-manager). When using the context manager, responses will fail if not all the configured responses are invoked. That should ensure that this code is working correctly without needed to inspect the logs.